### PR TITLE
Repeater test data

### PR DESCRIPTION
**Pull request by snopoke**
Send more realistic data when testing a repeater

http://manage.dimagi.com/default.asp?184464

### CODE REVIEW

File: corehq/apps/repeaters/repeater_generators.py [hunk 1]
```diff
@@ -45,6 +76,16 @@ def get_payload(self, repeat_record, payload_doc):
         data = payload_doc.get_json(lite=True)
         return json.dumps(data, cls=DjangoJSONEncoder)
 
+    def get_test_payload(self):
+        from casexml.apps.case.models import CommCareCase
+        return self.get_payload(
+                None,
```
Comment (dimagimon) [inline corehq/apps/repeaters/repeater_generators.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/repeaters/repeater_generators.py [hunk 2]
```diff
@@ -45,6 +76,16 @@ def get_payload(self, repeat_record, payload_doc):
...
+                CommCareCase(
+                        domain='demo-domain', type='case_type', name='Demo',
```
Comment (dimagimon) [inline corehq/apps/repeaters/repeater_generators.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/reports/tests/test_analytics.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
     get_all_form_details, get_form_details_for_xmlns, get_form_details_for_app_and_module, \
     get_form_details_for_app_and_xmlns, get_form_details_for_app
 from corehq.form_processor.interfaces.processor import FormProcessorInterface
-from corehq.form_processor.tests.utils import TestFormMetadata, get_simple_form_xml
+from corehq.form_processor.tests.utils import TestFormMetadata, get_simple_form_xml, get_simple_wrapped_form
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_analytics.py]: F401 'get_simple_form_xml' imported but unused

File: corehq/apps/repeaters/repeater_generators.py [hunk 4]
```diff
@@ -25,18 +36,38 @@ def get_payload(self, repeat_record, payload_doc):
     def get_headers(self, repeat_record, payload_doc):
         return {}
 
+    def get_test_payload(self):
+        return (
+            "<?xml version='1.0' ?>"
+            "<data id='test'>"
+            "<TestString>Test post from CommCareHQ on %s</TestString>"
+            "</data>" % datetime.utcnow()
+        )
```
Comment (benrudolph) [inline corehq/apps/repeaters/repeater_generators.py]: :100:

File: corehq/apps/domain/views.py [hunk 5]
```diff
@@ -532,27 +532,17 @@ def test_repeater(request, domain):
...
-                    case_name='test case',
-                ).as_string()
-            else:
-                return "<?xml version='1.0' ?><data id='test'><TestString>Test post from CommCareHQ on %s</TestString></data>" % \
-                       (datetime.datetime.utcnow())
-
-        fake_post = _stub(repeater_type)
+        format = format or RegisterGenerator.default_format_by_repeater(repeater_class)
+        generator_class = RegisterGenerator.generator_class_by_repeater_format(repeater_class, format)
+        fake_post = generator_class(repeater_class()).get_test_payload()
```
Comment (benrudolph) [inline corehq/apps/domain/views.py]: one day we should send a `json` body if the format is `json` rather than `xml`

Comment (benrudolph) [inline corehq/apps/domain/views.py]: actually it does look like some of the payloads are json. we may want to pass the format into `simple_post`: https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/post.py#L45

Comment (snopoke) [inline corehq/apps/domain/views.py]: :+1:

Comment (snopoke) [inline corehq/apps/domain/views.py]: https://github.com/dimagi/commcare-hq/commit/eca6f6b32b18a01b60bd8ee9e091113fa1e16cd3

**Comment by benrudolph** [comment]
:+1: nice, looks good once tests pass

**Comment by czue** [comment]
very nice

**Pull request merged**